src/ParameterLists.f90

```diff
@@ -2349,6 +2352,109 @@ FUNCTION has_ParamType(thisParam,name) RESULT(hasname)
     ENDFUNCTION has_ParamType
 !
 !-------------------------------------------------------------------------------
+!> @brief This subroutine will take a parameter list of parameter lists, where
+!>        each parameter list is a column to be added in the table.  The first
+!>        parameter list must be the maximum number of rows and must be uniquely
+!>        labeled.  The ordering of the parameters on the first list is the
+!>        order they will be written to the table. The following columns of
+!>        parameter lists must have a parameters with names that match those the
+!>        first column.  The value given to the parameter is arbitrary.
+!> @param thisParam The parameter list of parameter lists from which to create a
+!>        table.
+!> @param baseAddr The path used to extract the parameter list of parameter
+!>        lists.
+!> @param tablevals The 2-D string array that is allocated and returned.
+!>
+    SUBROUTINE convertTo2DStringArray_ParamType(thisParam,baseAddr,tablevals)
+      CLASS(ParamType),INTENT(IN) :: thisParam
+      TYPE(StringType),INTENT(IN) :: baseAddr
+      TYPE(StringType),ALLOCATABLE,INTENT(OUT) :: tablevals(:,:)
+      INTEGER(SIK) :: i,j,ncol,nrow
+      TYPE(StringType) :: tmpstr,addr,plstr
+      TYPE(StringType),ALLOCATABLE :: rownames(:)
+      TYPE(ParamType) :: colListPL,rowListPL
+      CLASS(ParamType),POINTER :: colListPLPtr,rowListPLPtr,tmpPLPtr,tmpPLPtrLast
+
+      !Initialize data
+      !sdkfmt=''
+      !IF(PL%has(baseAddr//'->sdkfmt')) &
+          !CALL PL%get(baseAddr//'->sdkfmt',sdkfmt)
+      !Loop over all columns, get the number of columns for the table
+      ncol=0
+      nrow=0
+      addr=baseAddr
+      colListPLPtr => NULL()
+      CALL thisParam%getSubPL(addr,colListPLPtr)
+      DO WHILE(ASSOCIATED(colListPLPtr))
+        ncol=ncol+1
+        CALL thisParam%getSubPL(addr,colListPLPtr)
+      ENDDO
+
+      addr='1'
```

Comment:
I answered this above, but I have a related question.  Do you think the requirement to have the column parameter lists named as 1, 2, 3, etc... is too restrictive?  I could have the code call `%getSubPL` and just loop over the columns that way.  Not sure if we need to add columns out of order, because then this would be more useful.  I do like removing some of the implicit structure if it isn't necessary.  Thoughts?

Reply:
Talked with Dan. Since the purpose of routine is not to convert general PL to a 2D table, but is exclusively used for printing state variables in a desired 2-D format, maybe state the purpose and limitations in the header of the routine, so others could be aware of it.